src/number.c

```diff
@@ -80,22 +81,18 @@ static val_t n_isNaN(struct v7 *v7, val_t this_obj, val_t args) {
 }
 
 V7_PRIVATE void init_number(struct v7 *v7) {
-  val_t num, ctor = v7_create_cfunction(Number_ctor);
+  val_t num = create_object(v7, v7->number_prototype);
+  val_t ctor = v7_create_cfunction(Number_ctor);
   unsigned int attrs = V7_PROPERTY_READ_ONLY | V7_PROPERTY_DONT_ENUM |
                        V7_PROPERTY_DONT_DELETE;
-  v7_exec(v7, &num, "function Number(v) { "
-          "  Object.defineProperty(this, '__ctor',"
-          "                        {value:__Number_ctor});"
-          "   return this.__ctor(v)}; "
-          "Number");
-  v7->number_prototype = v7_get(v7, num, "prototype", 9);
-  v7_get_property(num, "prototype", 9)->attributes |= V7_PROPERTY_READ_ONLY;
-
-  v7_set_property(v7, v7->global_object, "__Number_ctor", 13, attrs, ctor);
+  v7_set_property(v7, num, "", 0, V7_PROPERTY_HIDDEN, ctor);
+  v7_set_property(v7, num, "prototype", 9, attrs, v7->number_prototype);
+  v7_set_property(v7, v7->number_prototype, "constructor", 11, attrs, ctor);
```

Comment:
`s/ctor/num`

This would also fix the annoying `687` ecma test, annoying because the error message contains a pointer and thus it always diffs.